### PR TITLE
Block ability for project spaces to un-pause without paying their unpaid invoices

### PR DESCRIPTION
**Pull request by biyeun**
https://dimagi-dev.atlassian.net/browse/SAAS-10712

##### SUMMARY
Please start to review these changes as is. The UI-related part of this ticket I will do in a separate PR into this branch—otherwise, things might get a bit long. I want to keep all these changes together so that they are part of the same QA process.

Right now, this PR contains changes to the accounting admin UI that are present when the `Enable Accounting Testing Tools` feature flag is present:
- In `Trigger Invoice`, allow specifying the number of users to auto-generate a `DomainUserHistory` object before triggering an invoice.
- Creates a new view `Trigger Downgrade` which triggers the downgrade process (the exact same process run in `corehq.apps.accounting.tasks`), but only for the domain specified and no other domain. This means that QA will be able to trigger not only pauses but test the email reminder functionality in the downgrade process.

Recommended to review commit-by-commit

### CODE REVIEW

File: corehq/apps/accounting/utils/downgrade.py [hunk 1]
```diff
@@ -28,3 +40,115 @@ def send_downgrade_notice(invoice, context):
...
+        if not oldest_unpaid_invoice.is_customer_invoice:  # We do not automatically downgrade customer invoices
+            _downgrade_domain(subscription)
+            _send_downgrade_notice(oldest_unpaid_invoice, context)
+    elif days_ago == 58:
+        _send_downgrade_warning(oldest_unpaid_invoice, context)
+    elif days_ago == 30:
+        _send_overdue_notice(oldest_unpaid_invoice, context)
+
+
+def downgrade_eligible_domains():
```
Comment (millerdev) [inline corehq/apps/accounting/utils/downgrade.py]: What do you think of moving this to the top of the module? It appears to be the most important function in here.

**Review by millerdev — APPROVED** [review]
Nice cleanup/reorg.

**Comment by biyeun** [comment]
@millerdev thanks!

next round of changes here: https://github.com/dimagi/commcare-hq/pull/26998

File: corehq/apps/accounting/utils/invoicing.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    )
+
+
+def _get_unpaid_saas_invoices_in_downgrade_daterange(today):
+    return _get_all_unpaid_saas_invoices().filter(
+        date_due__lte=today - datetime.timedelta(days=1)
+    ).order_by('date_due').select_related('subscription__subscriber')
+
+
+def get_unpaid_invoices_over_threshold_by_domain(today, domain):
```
Comment (dannyroberts) [inline corehq/apps/accounting/utils/invoicing.py]: I realize your just moving this, and also that this is a base PR into which you're merging individually reviewed PRs, but in reviewing one of the PRs I got confused about what this function does and what it returns. From the name I though it returned unpaid invoices over some threshold, but it actually returns the oldest unpaid invoice whose amount is over some threshold, as well as the amount due in that invoice, as a tuple.

All that to say, might be worth changing the name of this function to be
`get_oldest_unpaid_invoice_over_threshold` or something like that

**Review by millerdev — APPROVED** [review]

File: corehq/apps/accounting/utils/invoicing.py (hunk 2 above)
Comment (biyeun) [reply corehq/apps/accounting/utils/invoicing.py:28]: @dannyroberts https://github.com/dimagi/commcare-hq/pull/27196

**Pull request merged**